test/util/oauthUtil.js

```diff
@@ -363,18 +363,36 @@ define(function(require) {
     });
 
     util.warpToUnixTime(getTime(opts.time));
-    util.mockGetLocationHash(client, opts.hashMock);
+    util.mockGetLocationHash(client, opts.hashMock || '');
+    var setLocationHashMock = util.mockSetLocationHash(client);
+    var replaceStateSpy = jasmine.createSpy('replaceState');
+    if (opts.noHistory) {
+      util.mockGetHistory(client);
+    } else {
+      util.mockGetHistory(client, {
+        replaceState: replaceStateSpy
+      });
+    }
     util.mockGetCookie(opts.oauthCookie);
     var setCookieMock = util.mockSetCookie();
 
-    return client.token.parseFromUrl()
+    return client.token.parseFromUrl(opts.directUrl)
       .then(function(res) {
         var expectedResp = opts.expectedResp;
         validateResponse(res, expectedResp);
 
         // The cookie should be deleted
         expect(setCookieMock).toHaveBeenCalledWith('okta-oauth-redirect-params=; path=/; ' +
           'expires=Thu, 01 Jan 1970 00:00:00 GMT;');
+
+        if (opts.directUrl) {
+          expect(setLocationHashMock).not.toHaveBeenCalled();
+          expect(replaceStateSpy).not.toHaveBeenCalled();
+        } else if (opts.noHistory) {
+          expect(setLocationHashMock).toHaveBeenCalledWith('');
+        } else {
+          expect(replaceStateSpy).toHaveBeenCalled();
```

Comment:
Does this line get invoked with the tests we have? And should we have an expectation here to verify that the url we are replacing it with is the url without the hash?